app-modules/assistant/database/migrations/2024_04_23_123415_create_ai_assistants_table.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2024, Canyon GBS LLC. All rights reserved.
+
+    Advising App™ is licensed under the Elastic License 2.0. For more details,
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+use Laravel\Pennant\Feature;
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+return new class () extends Migration {
+    public function up(): void
+    {
+        Schema::create('ai_assistants', function (Blueprint $table) {
+            $table->uuid('id')->primary();
+            $table->string('assistant_id')->nullable();
+
+            $table->string('name');
+            $table->string('type')->nullable();
+            $table->text('description')->nullable();
+            $table->longText('instructions')->nullable();
+            $table->longText('knowledge')->nullable();
+
+            $table->timestamps();
+            $table->softDeletes();
+        });
+
+        Schema::table('assistant_chats', function (Blueprint $table) {
+            $table->foreignUuid('ai_assistant_id')->nullable()->constrained('ai_assistants');
+        });
+
+        Feature::activate('custom-ai-assistants');
```

Comment:
Schema migrations should only contain schema migrations. Any data migration including feature activation or purging should be moved to a separate data migration.